src/app/Http/Controllers/UserController.php

```diff
@@ -19,19 +20,27 @@ public function __construct(UserFollowService $userFollowService)
 
     public function show(): View
     {
-        $user = User::find(auth()->id());
+        /** @var User $user */
+        $user = auth()->user();
 
-        if (!$user) {
-            return view('user.profile', ['posts' => collect()]);
-        }
         $posts = $user->userPosts()->latest()->paginate(4);
 
         return view('user.profile', ['user' => $user, 'posts' => $posts, 'userFollowService' => $this->userFollowService,]);
     }
 
     public function home(): View
     {
-        $posts = Post::latest()->paginate(4);
+        /** @var User $user */
+        $user = auth()->user();
+
+        $userIds = $user->followings
+            ->pluck('id')
+            ->push(Auth::id());
```

Comment:
why Auth::id()??

Reply:
maybe you can also use eager loading to optimize the query more.

Reply:
without Eager Loading, Laravel might execute additional database queries when you access related data, which can lead to the "N+1 problem

Reply:
yung about po sa Auth::id() sir, kinukuha ko rin po kasi yung post ni logged in user para mag didisplay din po sa home page hehe.

Reply:
Noted po sa eager loading, i will search about that po. Thank you po, sir!

Reply:
> yung about po sa Auth::id() sir, kinukuha ko rin po kasi yung post ni logged in user para mag didisplay din po sa home page hehe.

I know. why still use that. if you already done getting the user by using the `auth()->user()` you can use this instead.

`$user->id`

then from what I can see you can also add this as an accessor to the post model.

Reply:
Yes po, sir. Narealize ko rin po eventually hehe, I changed that code na po. Noted po sir thank you po!

Reply:
Create service layer for each method!